viewing_party/main.py

```diff
@@ -0,0 +1,218 @@
+    # print(user_data)
+    # print(user_data["watched"])
+
+    watched = user_data["watched"]
+    genre_freq_dict = calculate_genre_freq(watched)
+    most_popular = 0
+    most_popular_genre = ""
+    # print("before the loop")
+    #what if there was an empty list 
+    if len(watched) == 0:
+        return None
+    for key,value in genre_freq_dict.items():
+        if value > most_popular:
+            most_popular = value
+            most_popular_genre = key  
+    # print("after the loop")             
+    return most_popular_genre
+            
+
+
+
+
+
+#WAVE 3 
+
+def get_unique_watched(user_data):
+# we need to compare the watched lists for both the
+    # user and the freinds 
+    # find which movies friends haven't watched 
+    # but the user has 
+    # return a list of dict that shows these movies? 
+    # also add in if the list is empty 
+    
+    # user_watched_movies = [] #will hold the value of ["title"]
+    friends_watched_movies = [] #will hold the value of ["title"] 
+    user_unique_titles = [] #movies that we ended up comparing and the user had watched
+    
+    # for movie in user_data["watched"]:
+    #     user_watched_movies.append(movie["title"])       
+
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched_movies.append(movie)
+    
+    
+    #need to loop through unique watched movies 
+    # for unique_movies in user_data["watched"]
+    for unique_movies in user_data["watched"]:
+        if unique_movies not in friends_watched_movies:
+            user_unique_titles.append(unique_movies)
+    
+    return user_unique_titles
+
+
+#  to help if you want to use sets ??  
+# >>> {1, 2, 3, 4} - {2, 3}
+# {1, 4} 
+
+
+
+def get_friends_unique_watched(user_data):
+    # user_watched_movies = []
+    friends_watched_movies = []
+    unique_watched_movies = []
+
+    # for movie in user_data["watched"]:
+    #     user_watched_movies.append(movie["title"])
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched_movies.append(movie)
+    
+
+    
+    for movie in friends_watched_movies:
+        if movie not in user_data["watched"]:
+            if movie not in unique_watched_movies:
+                unique_watched_movies.append(movie)
+    
+    
+    
+    return unique_watched_movies    
+
+        
+    
+    # one friend has watched but the user has NOT watched
+    # for unique_movie in unique_watched_movies: 
+            
+
+
+
+
+#WAVE 4
+
+def get_available_recs(user_data):
+    # movies the user has not watched(can access via their sub) but a friend has watched
+    recommended_movies = []
+
+    # wave 3 function that returns the unique watched movie list
+    # why isn't there a test to check if the user hasn't watched the movie but their friends have? 
+    # unique_movie = get_friends_unique_watched(user_data) 
+    streaming_services = user_data["subscriptions"]
+
+
+
+
```

Comment:
This is not a big deal, but for future reference, the norm is to have less spacing between pieces of code.